### ISSUE TITLE
nothing drawn on ubuntu laptop with Integrated Intel® HD Graphics 4000 for gloo

### ISSUE THREAD

**Issue by matthew-egbert** [glumpy/glumpy#62]
I wrote a glumpy based visualization that works fine on my desktop, but only shows a blank screen on my laptop. Further testing shows that the app-*.py examples seem to work fine, but all of the gloo-*.py examples have the same problem of not showing anything but a window filled in a solid color. (The color appears to be determined by the clear color of the app. e.g. gloo-cube.py presents a grey window rather than a black one, presumably thanks to the line.

window = app.Window(width=1024, height=1024, **color=(0.30, 0.30, 0.35, 1.00)**)

I am running ubuntu 14.04 on both devices, but there appears to be a different version of GL reported by GLFW.

Desktop (which works) reports
[i] Using GLFW (GL 4.5)

Laptop  (which doesn't work) reports
[i] Using GLFW (GL 3.0)

I wonder if that is the problem? Any suggestions much appreciated.

**Comment by rougier** [comment]
What about the `gloo-texture-1D.py` and `gloo-texture-2D.py` ? Also black (or grey) ?

**Comment by matthew-egbert** [comment]
Yes, those both just put up a black window. I don't get any error message.

**Comment by matthew-egbert** [comment]
I think it has something to do with the opengl version. Is there a way in glumpy to specifically request a opengl context of a certain version?

**Comment by matthew-egbert** [comment]
Something similar to glutInitContextVersion(3,3);

**Comment by matthew-egbert** [comment]
I found these glumpy command line arguments, but none of them seem to help.

  --gl-api {GL,ES}      GL API
  --gl-profile {none,core,compatibility}
                        GL context profile (only relevant for GL > 3.0)
  --gl-version GL_VERSION
                        GL version

**Comment by rougier** [comment]
Can you also try the `colormaps.py` ? And by the way, does `glxgears` works ok ?
Sorry for all the test but I need more information to find the cause.

**Comment by matthew-egbert** [comment]
colormaps.py just makes a large black window, and glxgears works fine.
Thanks for looking into this!

**Comment by rougier** [comment]
Sorry for the delay. Can you also try: https://github.com/glumpy/glumpy/blob/master/examples/tutorial/quad-glut.py

This one does not use glumpy at all. We'll see if glumpy is responsible or not.

**Comment by matthew-egbert** [comment]
No, it doesn't work. On my desktop I get colorful window, on my laptop just
a blank black window.

**Comment by rougier** [comment]
Weird. Do you ever get an python/OpenGL to run on you machine ?
Maybe you can give a try at vispy (vispy.org) to see if you get the same kind of problem (I'm bit clueless at the moment)

**Comment by matthew-egbert** [comment]
Yes,

I don't have any problems with OpenGL and python in general.

As I wrote in an earlier post, my hunch is that the problem has to do
specifically with the openGL version and GLSL-style / modern-pipeline
openGL.

M

**Comment by rougier** [comment]
What is weird is that OpenGL does not complain about missing feature or methods. PyOpenGL should at least throw some warnings or errors if shaders were not accessible. I think you may need to post your problem on PyOpenGL mailing list with the glut example.

**Comment by rougier** [comment]
Any progress on that ? Could you post here you configuration ?

**Comment by matthew-egbert** [comment]
I recently posted a description of my problem to the pyopengl-users mailing
list here:

https://sourceforge.net/p/pyopengl/mailman/message/34834316/

but no response as of yet. I would love to resolve this problem, but I
don't know what else to try.

Cheers,
Matthew
